client/lib/sites-list/list.js

```diff
@@ -375,7 +375,7 @@ SitesList.prototype.getSite = function( siteID ) {
 		// clashes between a domain redirect and a Jetpack site, as well as domains
```

Comment:
I think it would make sense to move the comment as well to the new function. Since it gives it a bit of a better idea why we are doing the comparison.